openwisp_monitoring/notifications/admin.py

```diff
@@ -57,11 +77,87 @@ def mark_as_read(self, request, queryset):
 
     mark_as_read.short_description = _('Mark selected notifications as read')
 
+    def actor_object_link(self, obj):
+        try:
+            url = reverse(
+                'admin:{0}_{1}_change'.format(
+                    obj.actor_content_type.app_label, obj.actor_content_type.model
+                ),
+                args=(obj.actor_object_id,),
+            )
+            return format_html(
+                '<a href="{url}" id="actor-object-url">{id}</a>',
+                url=url,
+                id=obj.actor_object_id,
+            )
+        except NoReverseMatch:
+            return obj.actor_object_id
+        except AttributeError:
+            return '-'
+
+    actor_object_link.short_description = _('Actor Object')
+
+    def action_object_link(self, obj):
+        try:
+            url = reverse(
+                'admin:{0}_{1}_change'.format(
+                    obj.action_object_content_type.app_label,
+                    obj.action_object_content_type.model,
+                ),
+                args=(obj.action_object_object_id,),
+            )
+            return format_html(
+                '<a href="{url}" id="action-object-url">{id}</a>',
+                url=url,
+                id=obj.action_object_object_id,
+            )
+        except NoReverseMatch:
+            return obj.action_object_object_id
+        except AttributeError:
+            return '-'
+
+    action_object_link.short_description = _('Action Object')
+
+    def target_object_link(self, obj):
+        try:
+            url = reverse(
+                'admin:{0}_{1}_change'.format(
+                    obj.target_content_type.app_label, obj.target_content_type.model
+                ),
+                args=(obj.target_object_id,),
+            )
+            return format_html(
+                '<a href="{url}" id="target-object-url">{id}</a>',
+                url=url,
+                id=obj.target_object_id,
+            )
+        except NoReverseMatch:
+            return obj.target_object_id
+        except AttributeError:
+            return '-'
+
+    target_object_link.short_description = _('Target Object')
+
+    def related_object(self, obj):
+        target_object = self.target_object_link(obj)
+        if isinstance(target_object, SafeText):
+            url = self.target_object_link(obj).split('"')[1]
+            return format_html(
+                '<a href="{url}" id="related-object-url">{content_type} | {name}</a>',
```

Comment:
This is the anchor text I see:

`config | device | Cap-Dev-01`

But this would be more user friendly:

`device: Cap-Dev-01`

You should be able to get the device model name using `content_type.model` or something similar.